src/arkode/arkode_ls.c

```diff
@@ -525,18 +541,26 @@ int arkLSSetMassLinearSolver(void* arkode_mem, SUNLinearSolver LS, SUNMatrix M,
   ===============================================================*/
 
 /*---------------------------------------------------------------
-  arkLSSetJacFn specifies the Jacobian function.
+  ARKodeSetJacFn specifies the Jacobian function.
   ---------------------------------------------------------------*/
-int arkLSSetJacFn(void* arkode_mem, ARKLsJacFn jac)
+int ARKodeSetJacFn(void* arkode_mem, ARKLsJacFn jac)
 {
   ARKodeMem ark_mem;
   ARKLsMem arkls_mem;
   int retval;
 
-  /* access ARKLsMem structure */
-  retval = arkLs_AccessLMem(arkode_mem, __func__, &ark_mem, &arkls_mem);
+  /* access ARKodeMem and ARKLsMem structures */
+  retval = arkLs_AccessARKODELMem(arkode_mem, __func__, &ark_mem, &arkls_mem);
   if (retval != ARK_SUCCESS) { return (retval); }
 
+  /* Guard against use for time steppers that do not need an algebraic solver */
+  if (!ark_mem->step_supports_algebraic)
```

Comment:
I think this question relates to a discussion we had back in the fall regarding error handling, wherein we decided that it was preferable for a function to segfault on a user error (like here) then to insert a pile of extra checks and then eventually return an error that is harder for users to track down.

In some sense, I think that this whole `step_supports_X` approach ends up adding extra code bloat when a simple segfault would be cleaner, but I went with this because it was specifically requested during the code review.  I would like us to have a more thorough discussion of this topic in our meeting on Tuesday.